tests/test_rope_dail_vs_hf.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from composer.core.precision import get_precision_context
+
+from llmfoundry.models.layers.attention import is_flash_v2_installed
+
+if is_flash_v2_installed():
+    from flash_attn.layers.rotary import RotaryEmbedding as DAILRotaryEmbedding
+from omegaconf import OmegaConf as om
+from transformers.models.llama.modeling_llama import \
+    LlamaDynamicNTKScalingRotaryEmbedding as HFDynamicNTKScalingRotaryEmbedding
+from transformers.models.llama.modeling_llama import \
+    LlamaLinearScalingRotaryEmbedding as HFLinearScalingRotaryEmbedding
+from transformers.models.llama.modeling_llama import \
+    LlamaRotaryEmbedding as HFRotaryEmbedding
+
+
+def gen_rotary_embedding(rope_head_dim: int, pos_emb_config: dict,
```

Comment:
this function looks like it can be imported from the one you use in the actual code. if so, lets do that

Reply:
I did not originally import it because I wanted this test to be as independent of the actual code as possible (since this only checks if dail and hf implementations produce the same results). However, importing the function will also test the function indirectly, so I will import it now. 
**tldr**: Done